src/pages/dogs/ReportDogPage/ReportDogPage.tsx

```diff
@@ -0,0 +1,302 @@
+          dispatch({ type: "ADD_NEW_REPORT", payload: json.data });
+          setTimeout(() => {
+            // wait before navigating to results page in order to show the success/error toast
+            navigateToResultsPage({ base64Image });
+          }, 2000);
+        } else {
+          setRequestStatus("error");
+          setIsLoading(false);
+        }
+      } catch (error) {
+        setRequestStatus("error");
+        setIsLoading(false);
+        console.error(error); // eslint-disable-line
+      }
+    };
+
+    const { reportPage } = AppTexts;
+    const { request } = reportPage;
+
+    const successMessage =
+      dogType === DogType.LOST
+        ? request.success.reportedLost
+        : request.success.reportedFound;
+
+    const alertText =
+      requestStatus === "error" ? request.error : successMessage;
+
+    return (
+      <>
+        <MatchingReportModal
+          matchingReports={matchingReports}
+          isModalOpen={isModalOpen}
+          setIsModalOpen={setIsModalOpen}
+          dogType={dogType}
+        />
+        <PageContainer>
+          <Box sx={styles.root}>
+            <PageTitle text={title} />
+            <Snackbar
+              open={!!requestStatus}
+              autoHideDuration={6000}
+              onClose={handleCloseError}
+            >
+              <Alert
+                onClose={handleCloseError}
+                severity={requestStatus as AlertColor}
+                sx={styles.alert}
+              >
+                {alertText}
+                <br />
+                {alertText === successMessage && request.success.redirect}
+              </Alert>
+            </Snackbar>
+            {isLoading ? (
+              <CircularProgress />
+            ) : (
+              <>
+                <DogPhoto
+                  onSelectImage={onSelectImage}
+                  selectedImageUrl={selectedImageUrl}
+                  clearSelection={clearSelection}
+                  isError={isMissingImage}
+                />
+                {getInputsData(dogType).map((inputField) => {
+                  const selectedInput = inputs[inputField.name];
+                  const isSelectType = [
+                    // represents inputs with type="select"
+                    "dogSize",
+                    "dogSex",
+                    "ageGroup",
+                  ].includes(inputField.name);
+
+                  return (
+                    <div key={inputField.name} style={{ width: "100%" }}>
+                      {isSelectType ? (
+                        <SelectInputField
+                          options={inputField.options!}
+                          label={inputField.label}
+                          onChange={inputField.onChange}
+                          error={inputField.error}
+                          // @ts-expect-error
+                          value={selectedInput?.value!}
+                        />
+                      ) : inputField.name === "date" ? (
```

Comment:
We could also save the component type on the input, and save it's props and then render here on demand.
Something like 
```jsx
<input.component {...input.props, key="somekey"} />
```

But it's anyway much better than the 400+ lines file we had earlier. Thanks 🙏